composer/callbacks/checkpoint_saver.py

```diff
@@ -321,6 +328,31 @@ def __init__(
 
         self.start_batch = None
 
+        self.remote_uploader = None
+        backend, _, _ = parse_uri(save_folder)
+        self.remote_uploader_futures: List[List[Future]] = []
+        self.symlink_file_tasks: List[tuple[str, str]] = []
+        self.this_rank_saves_remote_symlinks: bool = False
+        self.tmp_dir_for_symlink = tempfile.TemporaryDirectory()
+        self.num_concurrent_uploads = num_concurrent_uploads
+
+        if backend != '':
+            if backend == 'wandb':
+                raise NotImplementedError(
+                    f'There is no implementation for WandB via URI. Please use '
+                    'WandBLogger with log_artifacts set to True',
+                )
+            elif backend not in ['s3', 'oci', 'gs', 'azure', 'dbfs']:
+                raise NotImplementedError(
+                    f'There is no implementation for the cloud backend {backend} via URI. Please use '
+                    'one of the supported RemoteUploaderDownloader object stores',
+                )
+            self.remote_uploader = RemoteUploader(
+                remote_folder=save_folder,
+                num_concurrent_uploads=self.num_concurrent_uploads,
+            )
```

Comment:
wait so... what actualyl is supported as a backend?

Reply:
it supports `['s3', 'oci', 'gs', 'azure', 'dbfs']`. Did i miss anything?